### PR TITLE
Add pkgs/check-spec-patches.sh to validate existence of patches.

### PR DESCRIPTION
**Pull request by jackorp**

### CODE REVIEW

File: pkgs/check-spec-patches.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+if [ -z "$file" ]; then
+	die "No specfile supplied"
+fi
+
+sources=""
+applied=""
+existing=""
+
+# Get patches that are packaged
+sources=$(grep -e "^Patch[0-9]*:" $file)
```
Comment (pvalena) [inline pkgs/check-spec-patches.sh]: This is better handled by spectool, isn't it?

**Comment by pvalena** [comment]
I'm sorry, but as it is I think it should be either rewritten in Ruby, or the parsing should be rewritten into Bash/Zsh ... what do you think is better?

File: pkgs/check-spec-patches.sh (hunk 1 above)
Comment (jackorp) [reply pkgs/check-spec-patches.sh]: I wasn't aware of such a tool, it seems to do the job in a parseable way.
But it fails with fedora/rawhide ruby for me (though fedpkg doesn't complain and can create srpm)...
```console
$ spectool --patches ./ruby.spec 
error: ./ruby.spec: line 112: failed to load macro file /home/jprokop/rpmbuild/SOURCES/macros.ruby
RPM Failed to parse spec file.
```

**Comment by jackorp** [comment]
> I'm sorry, but as it is I think it should be either rewritten in Ruby, or the parsing should be rewritten into Bash/Zsh ... what do you think is better?

I am personally leaning towards Ruby, the reason for the way it is now is the ease of data transform and comparison.

**Pull request closed without merging**